src/ui/shared/error-display/ErrorMessage.tsx

```diff
@@ -0,0 +1,168 @@
+import { BottomSheetDialog } from 'src/ui/ui-kit/ModalDialogs/BottomSheetDialog';
+import { DialogTitle } from 'src/ui/ui-kit/ModalDialogs/DialogTitle';
+import type { HTMLDialogElementInterface } from 'src/ui/ui-kit/ModalDialogs/HTMLDialogElementInterface';
+import { Button } from 'src/ui/ui-kit/Button';
+import { UnstyledAnchor } from 'src/ui/ui-kit/UnstyledAnchor';
+import { HStack } from 'src/ui/ui-kit/HStack';
+import { UIText } from 'src/ui/ui-kit/UIText';
+import { VStack } from 'src/ui/ui-kit/VStack';
+import { PortalToRootNode } from 'src/ui/components/PortalToRootNode';
+import { getError } from 'get-error';
+import { useCopyToClipboard } from '../useCopyToClipboard';
+import type { ParsedError } from './parseError';
+import { parseError } from './parseError';
+
+function ErrorDetails({ error }: { error: ParsedError }) {
+  const { handleCopy, isSuccess } = useCopyToClipboard({ text: error.message });
+  return (
+    <>
+      <div
+        style={{
+          color: 'var(--negative-500)',
+          overflowWrap: 'break-word',
+          maxHeight: '50vh',
+          overflowY: 'auto',
+        }}
+      >
+        {error.message}
+      </div>
+      <HStack gap={12} style={{ gridTemplateColumns: '1fr 1fr' }}>
+        <Button kind="regular" value="cancel" onClick={handleCopy}>
+          {isSuccess ? (
+            <HStack
+              gap={8}
+              alignItems="center"
+              style={{ width: 'max-content', marginInline: 'auto' }}
+            >
+              <CheckIcon
+                style={{
+                  display: 'block',
+                  width: 20,
+                  height: 20,
+                  color: 'var(--positive-500)',
+                }}
+              />
+              <span>Copied</span>
+            </HStack>
+          ) : (
+            <span>Copy</span>
+          )}
+        </Button>
+        <Button
+          as={UnstyledAnchor}
+          kind="primary"
+          style={{ paddingInline: 16 }}
+        >
+          Contact Support
+        </Button>
+      </HStack>
+    </>
+  );
+}
+
+export function ErrorMessage({ error }: { error: Error }) {
+  const parsed = parseError(error);
+  const dialogRef = useRef<HTMLDialogElementInterface>(null);
+  if (error.message === 'DeniedByUser') {
+    return null;
+  }
+  return (
+    <>
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          alignItems: 'center',
+          border: '2px solid var(--negative-300)',
+          padding: 8,
+          borderRadius: 16,
+          color: 'var(--negative-500)',
+        }}
+      >
+        <UIText
+          kind="body/regular"
+          style={{
+            display: '-webkit-box',
+            WebkitLineClamp: 2,
+            WebkitBoxOrient: 'vertical',
+            overflow: 'hidden',
+          }}
+        >
+          {parsed.display || parsed.message}
+        </UIText>
+        <Button
```

Comment:
With this arrow icon it loos like the whole panel is clickable. But this is probably a question to design